### PR TITLE
Replace spacing near scroll views with overscroll (COMMUNTIY)

### PR DESCRIPTION
**Pull request by fynngodau**
As previously discussed in https://github.com/corona-warn-app/cwa-app-android/pull/1882#discussion_r543602525, this PR removes white space near scroll views, mostly at the bottom, and instead adds that space inside the scroll view (so it scrolls with the content). This makes the app feel a lot better.

Here you can see some examples:

Before | After
---|---
Screenshot_1618984437| Screenshot_1618984392
Screenshot_1618986546 | Screenshot_1618986818
Screenshot_1618987264 | Screenshot_1618987190

These views have been fixed that way:

* App Information
* About the App
* Data Privacy
* Legal Notices
* Technical Hotline
* Statistics explanation
* Edit Places
* Edit People
* Settings
* Notifications
* Prioritized Background Activity
* Share Data
* Check In (onboarding / information)

The error report screen has also been fixed to remove the white space between the elevated, sticky content and the scrollable content. Note that here, I changed the total bottom padding of the scrollable content from `20dp` to `24dp` for consistency within the layout.

Additionally, I fixed the button in the check in layout to have the same margin to top and bottom as the Open Journal button in the Contact Journal onboarding / information screen.

---
Internal Tracking ID: [EXPOSUREAPP-6605](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6605)

### CODE REVIEW

**Comment by dsarkar** [comment]
@fynngodau Thanks. Internal Tracking ID: [EXPOSUREAPP-6605](https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-6605)


---
Corona-Warn-App Open Source Team

**Comment by fynngodau** [comment]
I added fixes for the following layouts, which I initially overlooked:

* Settings
* Notifications (notification settings)
* Prioritized Background Activity (battery optimization settings)
* Share Data (ppa settings)
* Detailed Information about Data Sharing

**Comment by fynngodau** [comment]
I undid the change to the Detailed Information about Data Sharing screen to avoid conflict with #2945.

**Comment by fynngodau** [comment]
One more screen:

* Check In onbaording

Also fixes the margin of the Check In onboarding Agree button to match that of the Open Journal button in the Contact Journal onboarding.

File: Corona-Warn-App/src/main/res/layout/fragment_information_about.xml [hunk 1]
```diff
@@ -24,7 +24,10 @@
             android:layout_width="0dp"
             android:layout_height="0dp"
             android:fillViewport="true"
-            app:layout_constraintBottom_toBottomOf="@+id/guideline_bottom"
+            android:paddingBottom="@dimen/spacing_small"
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/layout/fragment_information_about.xml:27]: I would align the bottom padding on this kind of fragments, I would go with `@dimen/spacing_normal` as it is here: https://github.com/corona-warn-app/cwa-app-android/pull/2884/files#diff-99108bde996a7dd32ce38fb5594a027efd59819f2b88d116dcc7b8f3eaced7ebR30

File: Corona-Warn-App/src/main/res/layout/fragment_information_contact.xml [hunk 2]
```diff
@@ -24,7 +24,10 @@
             android:layout_width="0dp"
             android:layout_height="0dp"
             android:fillViewport="true"
-            app:layout_constraintBottom_toBottomOf="@+id/guideline_bottom"
+            android:paddingBottom="@dimen/spacing_small"
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/layout/fragment_information_contact.xml:27]: Same as above.

File: Corona-Warn-App/src/main/res/layout/fragment_information_legal.xml [hunk 3]
```diff
@@ -31,9 +31,11 @@
         <ScrollView
             android:layout_width="0dp"
             android:layout_height="0dp"
-            android:fillViewport="true"
             android:focusable="true"
-            app:layout_constraintBottom_toBottomOf="@+id/guideline_bottom"
+            android:paddingBottom="@dimen/spacing_small"
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/layout/fragment_information_legal.xml:35]: Same as above.

File: Corona-Warn-App/src/main/res/layout/fragment_information_privacy.xml [hunk 4]
```diff
@@ -30,7 +30,9 @@
             android:layout_width="match_parent"
             android:layout_height="match_parent"
             android:fillViewport="true"
-            android:paddingBottom="@dimen/guideline_bottom">
+            android:paddingBottom="@dimen/guideline_bottom"
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/layout/fragment_information_privacy.xml:33]: Same as above.

File: Corona-Warn-App/src/main/res/layout/fragment_information_technical.xml [hunk 5]
```diff
@@ -30,7 +30,9 @@
             android:layout_width="match_parent"
             android:layout_height="match_parent"
             android:fillViewport="true"
-            android:paddingBottom="@dimen/guideline_bottom">
+            android:paddingBottom="@dimen/guideline_bottom"
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/layout/fragment_information_technical.xml:33]: Same as above.

File: Corona-Warn-App/src/main/res/layout/fragment_settings.xml [hunk 6]
```diff
@@ -43,7 +43,10 @@
             android:layout_width="0dp"
             android:layout_height="0dp"
             android:fillViewport="true"
-            app:layout_constraintBottom_toBottomOf="@+id/guideline_bottom"
+            android:paddingBottom="@dimen/spacing_small"
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/layout/fragment_settings.xml:46]: Same as above.

File: Corona-Warn-App/src/main/res/layout/fragment_settings_background_priority.xml [hunk 7]
```diff
@@ -31,7 +31,10 @@
             android:layout_width="0dp"
             android:layout_height="0dp"
             android:fillViewport="true"
-            app:layout_constraintBottom_toBottomOf="@+id/guideline_bottom"
+            android:paddingBottom="@dimen/spacing_small"
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/layout/fragment_settings_background_priority.xml:34]: Same as above.

File: Corona-Warn-App/src/main/res/layout/fragment_settings_notifications.xml [hunk 8]
```diff
@@ -32,7 +32,10 @@
             android:layout_width="0dp"
             android:layout_height="0dp"
             android:fillViewport="true"
-            app:layout_constraintBottom_toBottomOf="@+id/guideline_bottom"
+            android:paddingBottom="@dimen/spacing_small"
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/layout/fragment_settings_notifications.xml:35]: Same as above.

File: Corona-Warn-App/src/main/res/layout/fragment_settings_privacy_preserving_analytics.xml [hunk 9]
```diff
@@ -27,7 +27,10 @@
             android:id="@+id/scrollview"
             android:layout_width="0dp"
             android:layout_height="0dp"
-            app:layout_constraintBottom_toTopOf="@+id/guideline_bottom"
+            android:paddingBottom="@dimen/spacing_small"
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/layout/fragment_settings_privacy_preserving_analytics.xml:30]: Same as above.

File: Corona-Warn-App/src/main/res/layout/fragment_statistics_explanation.xml [hunk 10]
```diff
@@ -93,7 +93,10 @@
         <ScrollView
             android:layout_width="0dp"
             android:layout_height="0dp"
-            app:layout_constraintBottom_toBottomOf="@id/guideline_bottom"
+            android:paddingBottom="@dimen/spacing_small"
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/layout/fragment_statistics_explanation.xml:96]: Same as above. But there seems to be already more space.

File: Corona-Warn-App/src/main/res/layout/fragment_trace_location_onboarding.xml [hunk 11]
```diff
@@ -31,7 +32,9 @@
         android:id="@+id/check_in_onboarding_scroll_view"
         android:layout_width="match_parent"
         android:layout_height="0dp"
-        android:layout_marginBottom="20dp"
+        android:paddingBottom="@dimen/spacing_small"
```
Comment (harambasicluka) [inline Corona-Warn-App/src/main/res/layout/fragment_trace_location_onboarding.xml:35]: Seems fine, as the floating button is below.

**Review by harambasicluka — CHANGES_REQUESTED** [review]
Overall looks pretty good, but I would like to take this chance to align all bottom paddings :)

What do you think?

File: Corona-Warn-App/src/main/res/layout/fragment_statistics_explanation.xml (hunk 10 above)
Comment (fynngodau) [reply Corona-Warn-App/src/main/res/layout/fragment_statistics_explanation.xml:96]: The answer is here:

https://github.com/corona-warn-app/cwa-app-android/blob/e1ec946e09600826d16ea389aa8dd96f7c959129/Corona-Warn-App/src/main/res/layout/fragment_statistics_explanation.xml#L431

That's an odd margin… we should remove that before increasing the scroll view's padding.

**Comment by fynngodau** [comment]
@harambasicluka Agreed – I have pushed a change updating all the places you pointed out.

**Review by chiljamgossow — COMMENTED** [review]
Very nice! Thanks for fixing.

**Review by harambasicluka — APPROVED** [review]

**Pull request merged**